modules/service/main.tf

```diff
@@ -1,22 +1,20 @@
-locals {
-  service_name = var.service_name
-}
-
 module "iam_role" {
   source = "./iam_role"
 
-  service_name = local.service_name
+  service_name = var.service_name
 }
 
 resource "aws_service_discovery_service" "service_discovery" {
+  for_each = var.service_discovery_namespace_id == null ? {} : { single = "a" }
```

Comment:
Does `single = "a"` have any meaning here?

Reply:
Yes, this is similar to the "need a thing with a single element" in it to do optional thing for lists (but with maps!).

We refer to this `aws_service_discovery_service.service_discovery["single"]` elsewhere.

Reply:
Needs comment to clarify.